apps/prairielearn/src/schemas/schemas/infoAssessment.ts

```diff
@@ -0,0 +1,117 @@
+import { z } from 'zod';
+
+export default z
+  .object({
+    comment: z
+      .union([z.string(), z.array(z.any()), z.object({}).catchall(z.any())])
+      .describe('Arbitrary comment for reference purposes.')
+      .optional(),
+    uuid: z
+      .string()
+      .regex(
+        new RegExp('^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$'),
+      )
+      .describe('Unique identifier (UUID v4).'),
+    type: z.enum(['Homework', 'Exam']).describe('Type of the assessment.'),
```

Comment:
It seems like we will probably already have Zod types for a lot of these objects. It seems wise to break up these larger schemas into smaller types, and colocate the smaller types in the same file. This would require moving type declarations here too, if they are spread around the project.